src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -175,21 +178,81 @@ calculateClaimsFromFixSwapStream s periodicSchedule streamSchedule useAdjustedDa
   let f = fromSome s.calculationPeriodAmount.calculation.fixedRateSchedule
   pure $ createFixRatePaymentClaims streamSchedule periodicSchedule useAdjustedDatesForDcf f.initialValue issuerPaysLeg s.calculationPeriodAmount.calculation.dayCountFraction currency
 
+-- | Rule when one specific floating rate is provided for a stub period.
+getSingleStubRateFromFpml floatingRate = do
+  Some (Observe floatingRate.floatingRateIndex)
+
+-- | Rule when two specific floating rates are provided for a stub period.
+-- In this case, linear interpolation should be used, as specified here:
+-- https://www.isda.org/a/aWkgE/Linear-interpolation-04022022.pdf
+getInterpolatedStubRateFromFpml fr1 fr2 p cal convention = do
+  let
+    tenor1 = fromSome fr1.indexTenor
+    tenor2 = fromSome fr2.indexTenor
+    s = p.adjustedStartDate
+    p1 = adjustDate cal convention (addPeriod s tenor1)
+    p2 = adjustDate cal convention (addPeriod s tenor2)
+    tn = subDate p.adjustedEndDate s
+    t1 = subDate p1 s
+    t2 = subDate p2 s
+    d1 = tn - t1
+    d2 = t2 - tn
+    w1 = intToDecimal d2 / intToDecimal (d1 + d2)
+    w2 = 1.0 - w1
+
+  -- In the ISDA paper, the following expression is used for linear interpolation:
+  --Some (Observe (fr1.floatingRateIndex) + (Observe (fr2.floatingRateIndex) - Observe (fr1.floatingRateIndex)) * Const (intToDecimal (tn - t1) / intToDecimal (t2 - t1)))
+
+  -- This expression can be simplified to:
+  Some (Const w1 * Observe (fr1.floatingRateIndex) + Const w2 * Observe (fr2.floatingRateIndex))
+
+-- | Get the floating stub rate to be used for a stub period.
+getStubRateFloatingFromFpml floatingRates p cal convention = do
```

Comment:
Using a `case floatingRates of` pattern match here is more idiomatic, e.g. 

```
case floatingRates of
[rate] -> ...
[rate1, rate2] -> ... 
other -> error ""
```

Reply:
I agree, this looks much better, thanks!